bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,250 @@
+        *args:
+            Can be at most two of them.
+            a. The list of columns in  which case the starting arguments of the provided function callable in mapped to the columns in the list.
+                If not provide then it tries maps the name of the parameter to a column name of the RDF. Flags an error if not found.
+            b. The name of the Filter as a string.
+        extra_args: If the function needs any other parameters that are not columns and must be passed explicitly.
+    Returns:
+        self: The filtered instance of the RDF.
+    
+    Examples:
+    1. rdf.Filter("x>y")
+        Filter using an c-style expression
+    2. rdf.Filter(lambda x,y: x>y)
+        Filter using a python callable.
+    3. rdf.Filter(lambda x,y: x>y, ["y", "x"])
+        Filter using a python callable with column list passed explicitly.
+        This is equivalent to the filter "y>x".
+    """
+    if isinstance(func_or_str, str): # If string argument is passed. Invoke the Original Filters.
+        return rdf._OriginalFilter(func_or_str, *args)
+
+    if not callable(func_or_str): # The 1st argument is either a string or a python callable.
+        raise ValueError("Filter takes in either a string or a python callable as the first argument.")
+    
+    if len(args) > 2: 
+        raise ValueError("Filter takes in at most 4 arguments (func, cols, filter_name, extra_args).")
+    
+    func = func_or_str
+    jitter = FunctionJitter(rdf)
+    func.__annotations__['return'] = 'bool' # return type for Filters is bool # Note: You can keep double and Filter still works.
+
+    col_list = []
+    filter_name  = ""
+
+    if len(args) == 0:
+        pass
+    
+    if len(args) == 1:
+        if isinstance(args[0], list): 
+            col_list = args[0]
+        elif isinstance(args[0], str):
+            filter_name = args[0]
+        else:
+            raise ValueError("Invalid input obtained.") # ! TODO: Make this error better
+    
+    if len(args) == 2:
+        if isinstance(args[0], list) and isinstance(args[1], str):
+            col_list = args[0]; filter_name = args[1]
+        elif isinstance(args[1], list) and isinstance(args[0], str):
+            col_list = args[1]; filter_name = args[0]
+        else:
+            raise ValueError("Invalid input obtained.") # ! TODO: Make this error better
+    
+    func_call = jitter.jit_function(func, col_list, extra_args)
+    return rdf._OriginalFilter("Numba::" + func_call, filter_name)
+
+
+# This class is still to undergo some restructuring thus the docstring may not be complete.
+# The class will be fully realized after the Define has been Pythonized.
+class FunctionJitter:
+
+    function_cache = {} # Variable to store previous functions so as to not rejit.
+    lambda_function_counter = 0 # Counter to name the lambda functions
+    
+    def __init__(self, rdf = None) -> None:
+        self.rdf = rdf
+        self.col_names = None if rdf is None else rdf.GetColumnNames()
```

Comment:
All attributes of the class instance should be defined at class creation, for better code readability. There are many attributes defined much later in the class definition, e.g. `self.func_args`, `self.params`. They should either be defined already here or not used as instance attributes but maybe as function return values?

An example to start could be:
```suggestion
    def __init__(self, rdf = None) -> None:
        self.rdf: Optional[ROOT.RDataFrame] = rdf
        self.col_names: Optional[List[str]] = None if rdf is None else rdf.GetColumnNames()
        self.func: Optional[Callable] = None
        ...
```

Although keep in mind that I still have doubts about the presence of `rdf` in this `__init__` method

Reply:
Might need to discuss this but 
Considering that this class is going to be internalised to the RDF Pythonizations... ```rdf``` can be in the ``` __init__```?
     An alternative might be to pass ```rdf``` to the ```find_type``` method.

Reply:
If the `rdf` parameter is a fundamental component of the class, then it should not be optional, and the class name and docstring should somehow refer to this fact, perhaps calling it `RDFPythonFunctionJitter` or something along these lines

Reply:
An alternative would be to rename ``find_type``` to ```find_type_from_rdf```.
Might leave space in case the class ever finds need to be reused.

Reply:
I think that since we're starting this PR for RDF purposes only, we can assume for now that the class will be used within RDF. So I see no problem with making `rdf` a positional argument. The class can be generalised in the future if needed